### PR TITLE
[Chat][Service] Participants list from SDK to Redux State as Map

### PR DESCRIPTION
**Pull request by dhiraj**
## Purpose

* Gets the participants list from the SDk into our Redux store

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Run unit tests in `ChatActionHandlerUnitTest`
* Run unit tests in `ParticipantsReducerUnitTest`

## What to Check
Verify that the following are valid:
* requestChatParticipants() should be called when onChatInitialized() and list of participants should be sent to ChatServiceListener.handleInfoModel()

### CODE REVIEW

**Comment by dhiraj** [comment]
@microsoft-github-policy-service agree [company="Traversient Inc"]

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/sdk/ChatActionHandler.kt [hunk 1]
```diff
@@ -102,6 +102,12 @@ internal class ChatActionHandler(private val chatService: ChatService) {
             val error = ChatStateError(errorCode = ErrorCode.CHAT_START_EVENT_NOTIFICATIONS_FAILED)
             dispatch(ErrorAction.ChatStateErrorOccurred(chatStateError = error))
         }
+        try {
+            chatService.requestChatParticipants()
+        } catch (ex: Exception) {
+            val error = ChatStateError(errorCode = ErrorCode.CHAT_REQUEST_PARTICIPANTS_FAILED)
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/sdk/ChatActionHandler.kt:108]: can we say CHAT_REQUEST_PARTICIPANTS_FETCH_FAILED?

Comment (dhiraj) [reply azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/sdk/ChatActionHandler.kt]: Renamed in https://github.com/Azure/communication-ui-library-android/pull/494/commits/f7c9186588ba0c2dba29a4b50ae3f9b14c532b34

**Review by ShaunaSong — APPROVED** [review]

**Review by v-loalbert — APPROVED** [review]
👍

**Review by ahammer — APPROVED** [review]

**Pull request merged**